ui/js/controllers/perf/compare.js

```diff
@@ -1,10 +1,34 @@
 "use strict";
 
+perf.value('compareDefaultTimeRange', 86400 * 2);
+
+perf.factory('compareFactory',[
+    '$rootScope', 'ThResultSetModel',
+    function ($rootScope, ThResultSetModel) {
+        return {
+            verifyRevision : function(project, revision, rsid){
+                return ThResultSetModel.getResultSetsFromRevision(project.name, revision).then(
+                    function(resultSets) {
+                        var resultSet = resultSets[0];
+                        //TODO: this is a bit hacky to pass in 'original' as a text string
+                        if (rsid === 'original') {
+                            $rootScope.originalResultSet = resultSet;
+                        } else {
+                            $rootScope.newResultSet = resultSet;
+                        }
+                    },
+                    function(error) {
+                        $rotScope.errors.push(error);
```

Comment:
typo: $rotScope -> $rootScope

Incidentally it's not great programming practice to use $rootScope, as it's basically the same as using global variables (see e.g.: http://c2.com/cgi/wiki?GlobalVariablesAreBad) Preferably this would be refactored into a simple function which took a set of revision/project information and returned the corresponding resultsets in an easy-to-digest form, but that might be a bit of a distraction, so we can just keep going with the current approach.